src/transform.utils.ts

```diff
@@ -0,0 +1,95 @@
+export const transformQuery = (query: string, parameters?: any[]): [string, number] => {
+  const quoteCharacters = ['\'', '"']
+
+  let newQueryString = ''
+  let currentQuote = null
+  let numberOfParametersInQueryString = 0
+  let extendedParameterCount = 0
+
+  for (let i = 0; i < query.length; i += 1) {
+    const currentCharacter = query[i]
+    const currentCharacterEscaped = i !== 0 && query[i - 1] === '\\'
+
+    if (currentCharacter === '?' && !currentQuote) {
+      const parameter = parameters![numberOfParametersInQueryString]
+
+      if (Array.isArray(parameter)) {
+        const extendedParamNames: string[] = []
+
+        parameter.forEach(() => {
+          extendedParamNames.push(`:e_param_${extendedParameterCount}`)
```

Comment:
Maybe a .map here?

Reply:
You don't like this do you? :)

```
    if (currentCharacter === '?' && !currentQuote) {
      const parameter = parameters![numberOfParametersInQueryString]

      newQueryString += Array.isArray(parameter)
        ? parameter.map(() => `:e_param_${extendedParameterCount++}`).join(', ')
        : `:param_${numberOfParametersInQueryString}`

      numberOfParametersInQueryString += 1
    }
```

That is how I would normally approach that. Would you prefer?

```
 if (currentCharacter === '?' && !currentQuote) {
      const parameter = parameters![numberOfParametersInQueryString]

      if (Array.isArray(parameter)) {
        newQueryString += parameter
          .map(() => {
            extendedParameterCount += 1
            return `:e_param_${extendedParameterCount - 1}`
          })
          .join(', ')
      } else {
        const paramName = `:param_${numberOfParametersInQueryString}`
        newQueryString += paramName
      }

      numberOfParametersInQueryString += 1
    }
```

Reply:
Went with this:

```
    if (currentCharacter === '?' && !currentQuote) {
      const parameter = parameters![numberOfParametersInQueryString]

      if (Array.isArray(parameter)) {
        newQueryString += parameter
          .map(() => `:e_param_${extendedParameterCount}`)
          .join(', ')

        extendedParameterCount += 1
      } else {
        newQueryString += `:param_${numberOfParametersInQueryString}`
      }

      numberOfParametersInQueryString += 1
    }
```

Reply:
I like your second approach more, thanks.